lms/djangoapps/badges/api/serializers.py

```diff
@@ -28,3 +31,33 @@ class BadgeAssertionSerializer(serializers.ModelSerializer):
     class Meta:
         model = BadgeAssertion
         fields = ('badge_class', 'image_url', 'assertion_url', 'created')
+
+
+class BadgeUserSerializer(serializers.ModelSerializer):
+    """
+    Serializer for the BadgeAssertion model.
+    """
+    name = serializers.CharField(source='profile.name')
+
+    class Meta:
+        model = User
+        fields = ('username', 'name')
+    
+    def to_representation(self, instance):
```

Comment:
I don't think its good idea to override `to_representation` for populating `profile_image_url` field I think you should create a field `profile_image_url` in serializer and then write `get_profile_image_url` method to populate it.